pandas/_libs/tslibs/nattype.pyx

```diff
@@ -484,6 +484,17 @@ class NaTType(_NaT):
             - 'raise' will raise an AmbiguousTimeError for an ambiguous time
 
             .. versionadded:: 0.24.0
+        nonexistent : 'shift', 'NaT', default 'raise'
```

Comment:
if any way to share these doc-strings would be great, maybe templating?

Reply:
@jbrockmendel do you happen to know why for some of these NaT methods we aren't just passing is `Timestamp.[method].__doc__` instead of copying these docstrings?

Reply:
Because of the dependency structure.  We import NaT in timestznps but not the other way around.